### PR TITLE
Unix Domain Socket - fix bug causing write side closure of socket to be swallowed

### PR DESCRIPTION
**Pull request by longshorej**
Unix Domain Socket - If writing fails, we need to fail the `sent` promise with the exception so that it can bubble up appropriately. This PR does that.

Additionally, the use of `BroadcastHub` was preventing the sink cancelling the attached `Source`, so now using `Sink.fromGraph(Flow[ByteString]...)`.

### CODE REVIEW

**Comment by longshorej** [comment]
cc @huntc

File: unix-domain-socket/src/main/scala/akka/stream/alpakka/unixdomainsocket/scaladsl/UnixDomainSocket.scala [hunk 1]
```diff
@@ -251,47 +252,46 @@ object UnixDomainSocket extends ExtensionId[UnixDomainSocket] with ExtensionIdPr
...
-          sendSource
-            .expand { bytes =>
-              if (bytes.size <= sendBufferSize) {
-                List(bytes).toIterator
+
+    val sendSink = Sink.fromGraph(
+      Flow[ByteString]
+        .expand { bytes =>
+          if (bytes.size <= sendBufferSize) {
+            List(bytes).toIterator
```
Comment (viktorklang) [inline unix-domain-socket/src/main/scala/akka/stream/alpakka/unixdomainsocket/scaladsl/UnixDomainSocket.scala]: There's also: `Iterator.single(bytes)`

Comment (longshorej) [reply unix-domain-socket/src/main/scala/akka/stream/alpakka/unixdomainsocket/scaladsl/UnixDomainSocket.scala]: Thanks! - fixed that. Originally I tried to keep changes not related to the bug to a minimum but might as well. Also changed the collection type there to a `Vector` as we're appending.

I presume `Iterator.single` is special cased and performs a bit better?

Comment (viktorklang) [reply unix-domain-socket/src/main/scala/akka/stream/alpakka/unixdomainsocket/scaladsl/UnixDomainSocket.scala]: @longshorej Fear not! :) In Akka "Apply the Boy Scout Rule whenever you have the chance to." - https://github.com/akka/akka/blob/master/CONTRIBUTING.md#additional-guidelines

It does indeed have a performance difference: List(bytes).toIterator allocates both a new List, then allocates a new Iterator. Iterator.single() only allocates the Iterator.

**Comment by 2m** [comment]
Validation failures are due to the Bintray repo problems. Unix domain socket tests completed successfully.

**Pull request merged**